src/Auth/CookieAuthenticate.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Auth;
+
+use Cake\Auth\BaseAuthenticate;
+use Cake\Controller\ComponentRegistry;
+use Cake\Controller\Component\CookieComponent;
+use Cake\Controller\Exception\MissingComponentException;
+use Cake\Event\Event;
+use Cake\Network\Request;
+use Cake\Network\Response;
+use Cake\Routing\Router;
+
+/**
+ * Cookie-based adapter for AuthComponent.
+ *
+ * Provides the ability to authenticate using COOKIE
+ *
+ * ```
+ *    $this->Auth->config('authenticate', [
+ *        'Authenticate.Cookie' => [
+ *            'fields' => [
+ *                'identifier' => 'username',
+ *                'token' => 'password'
```

Comment:
Ideally we wouldn't be putting the password hash in the cookie, and instead some other value that isn't the password would go into the cookie.